src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,109 @@ def standardize_patient_name(
+    return countries
 
-    raw = [x for x in raw if x.isnumeric()]
-    raw = "".join(raw)
-    if len(raw) != 10:
-        raw = None
-    return raw
+
+def standardize_phone(raw: str, countries: List = [None, "US"]) -> str:
+    """
+    Given a phone number and optionally an associated FHIR resource and country
+    extraction function if able to parse the phone number return it in the E.164
+    standardard international format. If the phone number is not parseable return none.
+
+    Phone number parsing process:
+    A maximum of three attemtps are made to parse any phone number.
+    First, we try to parse the phone number without any additional country information.
+    If this succeeds then the phone number must have been provided in a standard
+    inernational format which is the ideal case. If the first attempt fails and a list
+    of country codes indicating possible countries of origin for the phone number has
+    been provided we attempt to parse the phone number using that additional country
+    information from the resource. Finally, in the case where the second attempt fails
+    or a list of countries has not been provided we make a final attempt to parse the
+    number assuming it is American.
+
+    :param str raw: Raw phone number to be standardized.
+    :param list countries: A list containing 2 letter ISO country codes for each country
+    extracted from resource of the phone number to be standardized that might indicate
+    it the phone numbers country of origin.
+    :return str standardized: The standardized phone number in E.164 format when the raw
+    phone number was succesfully parsed and an emptry string otherwise.
+    """
+
+    if countries != [None, "US"]:
+        countries = [None] + countries + ["US"]
+
+    standardized = ""
+    for country in countries:
+        try:
+            standardized = phonenumbers.parse(raw, country)
+            break
+        except phonenumbers.phonenumberutil.NumberParseException:
+            continue
+    if standardized != "":
+        standardized = str(
+            phonenumbers.format_number(
+                standardized, phonenumbers.PhoneNumberFormat.E164
+            )
+        )
+    return standardized
+
+
+def standardize_country(
+    raw: str, code_type: Literal["alpha_2", "alpha_3", "numeric"] = "alpha_2"
+) -> str:
+    """
+    Given a country return it in a standard form as specified by code_type.
+
+    :param str raw: Country to be standardized.
+    :param str code_type: A string equal to 'alpha_2', 'alpha_3', or 'numeric' to
+    specify which type of standard country identifier to generate.
+    :return str standard: Country in standardized form, or None if unable to
+    standardize.
+    """
+    standard = ""
+    raw = raw.strip().upper()
+    if len(raw) == 3:
+        standard = pycountry.countries.get(alpha_3=raw)
+    elif len(raw) == 2:
+        standard = pycountry.countries.get(alpha_2=raw)
+    elif len(raw) == 3:
```

Comment:
This `if` logic looks off as you're testing `len(raw) == 3` twice. Is that intentional?

Reply:
Nope, not intentional just a result of working too fast. I think the changes in [d5edfd9](https://github.com/CDCgov/prime-public-health-data-infrastructure/pull/123/commits/d5edfd93601691afb236388136e517254d8a4980) are a little more readable and resolve this issue.